sql/spatial.h

```diff
@@ -289,6 +289,7 @@ class Geometry
   virtual int geom_length(double *len, const char **end) const  { return -1; }
   virtual int area(double *ar, const char **end) const { return -1;}
   virtual int is_closed(int *closed) const { return -1; }
+  virtual int is_valid(int *valid) const { return -1; }
```

Comment:
I'm confused by the interface that accepts a pointer that seems to in most case be populated with the negated value of the return. Is there a reason both are needed? If not, just return a value.

Reply:
`is_valid` returns 0 if the function completed successfully (regardless the geometry is valid or not), and returns 1 otherwise (Any error happens inside). This same logic exist in all other functions like `is_closed`

Then we check in `Item_func_isvalid::val_int()`
```cpp
if(g->is_valid(&valid))
  return -1;
```
If it returns true (error happened) just return -1 in `Item_func_isvalid::val_int()`